cmd/zpool/zpool_vdev.c

```diff
@@ -1638,7 +1641,8 @@ construct_spec(nvlist_t *props, int argc, char **argv)
 		free(spares);
 	if (l2cache)
 		free(l2cache);
-	free(top);
+	if (top)
```

Comment:
So in fact, this patch should be doing the opposite... change if(l2cache) free(l2cache) to just free(l2cache). Likewise for spares.